drivers/kscan/kscan_stmpe811.c

```diff
@@ -0,0 +1,440 @@
+	if (ret < 0) {
+		LOG_ERR("failed to write TSC_CFG");
+		return ret;
+	}
+
+	/* Configure the Touch FIFO threshold: 1 points reading */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_FIFO_TH, 0x01);
+	if (ret < 0) {
+		LOG_ERR("failed to write FIFO_TH");
+		return ret;
+	}
+
+	/* Clear the FIFO memory content. */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_FIFO_STA, 0x01);
+	if (ret < 0) {
+		LOG_ERR("failed to write FIFO_STA");
+		return ret;
+	}
+
+	/* Put the FIFO back into operation mode  */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_FIFO_STA, 0x00);
+	if (ret < 0) {
+		LOG_ERR("failed to write FIFO_STA");
+		return ret;
+	}
+
+	/* Set the driving capability (limit) of the device for TSC pins: 50mA */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_TSC_I_DRIVE, 0x01);
+	if (ret < 0) {
+		LOG_ERR("failed to write TSC_I_DRIVE");
+		return ret;
+	}
+
+	/* Touch screen control configuration (enable TSC):
+	 * - No window tracking index
+	 * - XY acquisition mode
+	 */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_TSC_CTRL, 0x03);
+	if (ret < 0) {
+		LOG_ERR("failed to write TSC_CTRL");
+		return ret;
+	}
+
+	/*  Clear all the status pending bits if any */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_INT_STA, 0xFF);
+	if (ret < 0) {
+		LOG_ERR("failed to write INT_STA");
+		return ret;
+	}
+
+	k_work_init(&data->work, stmpe811_work_handler);
+
+#ifdef CONFIG_KSCAN_STMPE811_INTERRUPT
+	if (!device_is_ready(config->int_gpio.port)) {
+		LOG_ERR("Interrupt GPIO controller device not ready");
+		return -ENODEV;
+	}
+
+	ret = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
+	if (ret < 0) {
+		LOG_ERR("Could not configure interrupt GPIO pin");
+		return ret;
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&config->int_gpio,
+					    GPIO_INT_EDGE_TO_ACTIVE);
+	if (ret < 0) {
+		LOG_ERR("Could not configure interrupt GPIO.");
+		return ret;
+	}
+
+	/* Enable global interrupt on STMPE811 */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_INT_CTRL, 0x1);
+	if (ret < 0) {
+		LOG_ERR("failed to write INT_CTRL");
+		return ret;
+	}
+
+	/* Enable global FIFO_TH interrupt on STMPE811 */
+	ret = i2c_reg_write_byte_dt(&config->bus, STMPE811_REG_INT_EN, 0x2);
+	if (ret < 0) {
+		LOG_ERR("failed to write INT_EN");
+		return ret;
+	}
+	gpio_init_callback(&data->int_gpio_cb, stmpe811_isr_handler,
+			   BIT(config->int_gpio.pin));
+#else
+	k_timer_init(&data->timer, stmpe811_timer_handler, NULL);
+#endif
+
+	return ret;
+}
+
+static const struct kscan_driver_api stmpe811_driver_api = {
+	.config = stmpe811_configure,
+	.enable_callback = stmpe811_enable_callback,
+	.disable_callback = stmpe811_disable_callback,
+};
+
+#define STMPE811_INIT(index)									\
+	static const struct stmpe811_config stmpe811_config_##index = {	\
+		.bus = I2C_DT_SPEC_INST_GET(index),						\
+		IF_ENABLED(CONFIG_KSCAN_STMPE811_INTERRUPT,					\
+		(.int_gpio = GPIO_DT_SPEC_INST_GET(index, int_gpios),))		\
```

Comment:
Even when CONFIG_KSCAN_STMPE811_INTERRUPT=y, the `int-gpios` property is still `required: false`. Therefore it might be missing from the devicetree. What is the right behavior in that case? What if there are 2 devices, one with an interrupt GPIO connected to the SoC, and one without? It seems like the driver may be able to handle this case without a compile error.